enclave-runtime/src/top_pool/primitives.rs

```diff
@@ -256,6 +253,12 @@ pub trait TrustedOperationPool: Send + Sync {
 		hash: &TxHash<Self>,
 		shard: ShardIdentifier,
 	) -> Option<Arc<Self::InPoolOperation>>;
+
+	/// Notify the listener of top inclusion in sidechain block
+	fn on_block_created(&self, hashes: &[Self::Hash], block_hash: SidechainBlockHash);
+
+	/// Notify the RPC client of a state update
+	fn rpc_send_state(&self, hash: Self::Hash, state_encoded: Vec<u8>) -> Result<(), error::Error>;
```

Comment:
Extend the `TrustedOperationPool` trait, so we can abstract the concrete pool types away (these methods were impls on the concrete `BasicPool`)

Reply:
I was waiting for this!

Reply:
I was waiting for this!